pkg/ccl/changefeedccl/parquet_test.go

```diff
@@ -0,0 +1,249 @@
+		require.Equal(t, expected.(*tree.DDate).Date.UnixEpochDays(),
+			actual.(*tree.DDate).Date.UnixEpochDays())
+	case types.JsonFamily:
+		// Only the value of the json object matters, not that additional properties
+		require.Equal(t, expected.(*tree.DJSON).JSON.String(),
+			actual.(*tree.DJSON).JSON.String())
+	case types.FloatFamily:
+		if typ.Equal(types.Float4) && expected.(*tree.DFloat).String() != "NaN" {
+			// CRDB currently doesn't truncate non NAN float4's correctly, so this
+			// test does it manually :(
+			// https://github.com/cockroachdb/cockroach/issues/73743
+			e := float32(*expected.(*tree.DFloat))
+			a := float32(*expected.(*tree.DFloat))
+			require.Equal(t, e, a)
+		} else {
+			require.Equal(t, expected.String(), actual.String())
+		}
+	case types.DecimalFamily:
+		require.Equal(t, expected.String(), actual.String())
+
+	default:
+		require.Equal(t, expected, actual)
+	}
+}
+
+func TestBasicParquetTypes(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	dir, dirCleanupFn := testutils.TempDir(t)
+	defer dirCleanupFn()
+	dbName := "baz"
+	srv, db, _ := serverutils.StartServer(t, base.TestServerArgs{
+		// Test fails when run within a test tenant. More
+		// investigation is required. Tracked with #76378.
+		DisableDefaultTestTenant: true,
+		UseDatabase:              dbName,
+		ExternalIODir:            dir,
+	})
+	ctx := context.Background()
+	defer srv.Stopper().Stop(ctx)
+	sqlDB := sqlutils.MakeSQLRunner(db)
+	sqlDB.ExecMultiple(t, strings.Split(serverSetupStatements, ";")...)
+
+	sqlDB.Exec(t, fmt.Sprintf("CREATE DATABASE %s", dbName))
+
+	// instantiating an internal executor to easily get datums from the table
+	ie := srv.ExecutorConfig().(sql.ExecutorConfig).InternalExecutor
+
+	sqlDB.Exec(t, `CREATE TABLE foo (id INT PRIMARY KEY, city STRING)`)
+	sqlDB.Exec(t, `INSERT INTO foo VALUES (1, 'Berlin')`)
+	// sqlDB.Exec(t, `CREATE TABLE foo (i INT PRIMARY KEY, x STRING, y INT, z FLOAT NOT NULL, a BOOL,
+	// INDEX (y))`)
+	// sqlDB.Exec(t, `INSERT INTO foo VALUES (1, 'Alice', 3, 0.5032135844230652, true), (2, 'Bob',
+	// 2, CAST('nan' AS FLOAT),false),(3, 'Carl', 1, 0.5032135844230652,true),(4, 'Alex', 3, 14.3, NULL), (5,
+	// 'Bobby', 2, 3.4,false), (6, NULL, NULL, 4.5, NULL)`)
+
+	tests := []parquetTest{
+		{
+			filePrefix:     "basic",
+			stmt:           `CREATE CHANGEFEED for foo into 'nodelocal://0/basic' with initial_scan='only',format='parquet',key_in_value`,
+			validationStmt: `select * from foo`,
+		},
+	}
+
+	for _, test := range tests {
+		t.Logf("Test %s", test.filePrefix)
+		if test.prep != nil {
+			for _, cmd := range test.prep {
+				sqlDB.Exec(t, cmd)
+			}
+		}
+
+		sqlDB.Exec(t, test.stmt)
+		time.Sleep(3 * time.Second)
+		test.dir = dir
+		test.dbName = dbName
+		paths, err := filepath.Glob(filepath.Join(test.dir, test.filePrefix))
+		require.NoError(t, err)
+		require.Equal(t, 1, len(paths))
+		walkFn := func(path string, info fs.FileInfo, err error) error {
```

Comment:
it feels to me that the fact that you're walking directory, and writing all of this hand logic, that the processing of parquet events should have been *very* similar to cloud storage test fixture.  It also waits for files to be flushed; it walks directory, it reads all the data in, and then simply produces events in the correct order.
I strongly suspect you'd be able to do something like this and build up on the existing test feed implementation.

I'm not saying that this test (parquet_test) necessarily needs to go; perhaps it can be repurposed to test specific functionality in the parquet_sink (i.e. we have sink_webhook_test for exampe);

But the test based on reading data from the database, and then expecting certain values, can almost certainly be expressed using test feed implementation.